ush/HWP_tools.py

```diff
@@ -79,6 +108,18 @@ def copy_missing_restart(nwges_dir, hwp_non_avail_hours, hourly_hwpdir):
     return(restart_avail_hours, restart_nonavail_hours_test)
 
 def process_hwp(fcst_dates, hourly_hwpdir, cols, rows, intp_dir, rave_to_intp):
+    """Check the restart files.
+
+    Args:
+         fcst_dates: a list of forecasts (for production/ops/ebb=2, it is {current_day - 25 hours: current_day-1hours}, for ebb=1, it is current_day:current_day+24 hours)
+         hourly_hwpdir: the directory where the RESTART data is copied (nwges/HOURLY_HWP)
+         cols: hard coded dimension for the RRFS_NA_3km and RRFS_CONUS_3km domains:: cols = 2700 if predef_grid == 'RRFS_NA_3km' else 1092
+         rows: hard coded dimension for the RRFS_NA_3km and RRFS_CONUS_3km domains:: rows = 3950 if predef_grid == 'RRFS_NA_3km' else 1820
+         intp_dir: the working directory for smoke processing (${CYCLE_DIR}/process_smoke)
+         rave_to_intp: a string based on the grid name:: rave_to_intp = predef_grid+"intp"
+    Returns:
+         list of (hwp_ave_arr, xarr_hwp, totprcp_ave_arr, xarr_totprcp)
+    """

```

Comment:
@jordanschnell what are the meanings of:
hwp_ave_arr, xarr_hwp, totprcp_ave_arr, xarr_totprcp

Reply:
hwp_ave_arr = average (through time) of HWP_AVE in the available RESTART files
xarr_hwp = hwp_ave_arr converted to an xarray DataArray
totprcp_ave_arr = is the total precipitation vector (totprcp) reshaped to the 2D grid
xarr_totprcp = totprcp_ave_arr converted to an xarray DataArray